### PR TITLE
#24 rename package to `roave/backward-compatibility-check`

### LINKED ISSUE #24: Re-naming the package...?
I've noticed when explaining what the tool does, folks jump to conclusion that this is used for comparing HTTP/REST API versions. I wonder if renaming the package could alleviate this problem. Would be better to do it before we tag anything as naturally it's (ironically) a big BC break ;)

Not sure WHAT to suggest though for naming, feedback welcome on this.

### PR DESCRIPTION
**Pull request by Ocramius**
Fixes #24

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,10 @@ jobs:
       php: 7.2
       script: vendor/bin/infection
 
+    - stage: Verify BC Breaks
+      php: 7.2
+      script: bin/roave-backward-compatibility-check --from=master --to=HEAD
```
Comment (Ocramius) [inline .travis.yml]: Note: this will fail for now, as this patch does indeed lead to BC breaks (namespace changes)

File: README.md [hunk 2]
```diff
@@ -1,66 +1,54 @@
-# API Compare Tool
+# Roave Backward Compatibility Check
```
Comment (Ocramius) [inline README.md]: Overall, I aimed at a few things:

 1. cut to the point in the docs
 2. moved docs for the CLI command into the CLI command itself
 3. made the `roave` bit obnoxiously repetitive: I like throwing free time at OSS, but people can indeed type `roave` a few times.

File: src/Changes.php [hunk 3]
```diff
@@ -2,55 +2,53 @@
...
-     * @param Change[] $changes
-     * @return Changes
-     */
-    public static function fromArray(array $changes) : self
-    {
-        Assert::that($changes)->all()->isInstanceOf(Change::class);
-        $instance          = self::new();
-        $instance->changes = $changes;
-        return $instance;
+        if ($empty) {
```
Comment (Ocramius) [inline src/Changes.php]: I found (via `Kcachegrind`) that this was really trashing memory and CPU on this side (together with `fromArray()` and `mergeWith()`)

File: src/Changes.php [hunk 4]
```diff
@@ -2,55 +2,53 @@
...
         return $instance;
     }
 
-    public function withAddedChange(Change $change) : self
+    public function mergeWith(self $other) : self
     {
-        $new            = clone $this;
-        $new->changes[] = $change;
-        return $new;
+        if (! $other->changes) {
```
Comment (Ocramius) [inline src/Changes.php]: Looked useless at first, but saves us from loads of instantiations

File: src/DetectChanges/BCBreak/MethodBased/MethodBased.php [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+declare(strict_types=1);
+
+namespace Roave\BackwardCompatibility\DetectChanges\BCBreak\MethodBased;
+
+use Roave\BackwardCompatibility\Changes;
+use Roave\BetterReflection\Reflection\ReflectionMethod;
+
+interface MethodBased
+{
+    public function __invoke(ReflectionMethod $fromMethod, ReflectionMethod $toMethod) : Changes;
```
Comment (Ocramius) [inline src/DetectChanges/BCBreak/MethodBased/MethodBased.php]: All checkers are now functional interfaces - didn't like the repeated name, nor the `compare()` wording, and settled with using just the wording from the class name

**Comment by asgrim** [comment]
Build failures were due to BC breaks indeed, so merging.

**Pull request merged**